pkgs/by-name/st/stardust-xr-kiara/package.nix

```
@@ -0,0 +1,54 @@
+{
+  lib,
+  fetchFromGitHub,
+  fetchpatch,
+  writeScript,
+  unstableGitUpdater,
+  rustPlatform,
+  cmake,
+  makeWrapper,
+  pkg-config,
+  niri,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "stardust-xr-kiara";
+  version = "0-unstable-2024-07-07";
+
+  src = fetchFromGitHub {
+    owner = "stardustxr";
+    repo = "kiara";
+    rev = "7daaa0a2e3822d949e6c4abf93af159eae9a544a";
+    hash = "sha256-5j83e2kcCStPgbwAkr3OFjOpJIErXAPJ6z06BlmtuHE=";
+  };
+
+  NIRI_CONFIG = "${src}/src/niri_config.kdl";
+  STARDUST_RES_PREFIXES = "${src}/res";
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "stardust-xr-0.14.1" = "sha256-fmRb46s0Ec8wnoerBh4JCv1WKz2of1YW+YGwy0Gr/yQ=";
+      "stardust-xr-molecules-0.29.0" = "sha256-sXwzrh052DCo7Jj1waebqKVmX8J9VRj5DpeUcGq3W2k=";
+    };
+  };
+  nativeBuildInputs = [ makeWrapper ];
```

nit: unless we *need* the shell-based wrapper for conditionals, we should use the binary wrapper as it's much faster.

```suggestion
  nativeBuildInputs = [ makeBinaryWrapper ];
```